mergify_engine/tests/functional/actions/test_queue.py

```diff
@@ -2585,6 +2587,974 @@ async def test_queue_manual_merge(self):
+                        "status-success=continuous-integration/fast-ci",
+                    ],
+                },
+                {
+                    "name": "default",
+                    "conditions": [
+                        "status-success=continuous-integration/slow-ci",
+                    ],
+                },
+                {
+                    "name": "low-priority",
+                    "conditions": [
+                        "status-success=continuous-integration/slow-ci",
+                    ],
+                },
+            ],
+            "pull_request_rules": [
+                {
+                    "name": "Merge priority high",
+                    "conditions": [
+                        f"base={self.main_branch_name}",
+                        "label=queue-urgent",
+                    ],
+                    "actions": {"queue": {"name": "urgent"}},
+                },
+                {
+                    "name": "Merge default",
+                    "conditions": [
+                        f"base={self.main_branch_name}",
+                        "label=queue",
+                    ],
+                    "actions": {"queue": {"name": "default"}},
+                },
+                {
+                    "name": "Merge low",
+                    "conditions": [
+                        f"base={self.main_branch_name}",
+                        "label=queue-low",
+                    ],
+                    "actions": {"queue": {"name": "low-priority"}},
+                },
+            ],
+        }
+
+        await self.setup_repo(yaml.dump(rules))
+        queue_name = "false_queue_name"
+
+        r = await self.app.put(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queue/{queue_name}/freeze",
+            json=None,
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+        assert r.status_code == 422
+        assert r.json() == {
+            "detail": [
+                {
+                    "loc": ["body"],
+                    "msg": "field required",
+                    "type": "value_error.missing",
+                }
+            ]
+        }
+
+        freeze_payload = {"false_key": "test freeze reason"}
+
+        r = await self.app.put(
+            f"/v1/repos/{config.TESTING_ORGANIZATION_NAME}/{self.RECORD_CONFIG['repository_name']}/queue/{queue_name}/freeze",
+            json=freeze_payload,
+            headers={
+                "Authorization": f"bearer {self.api_key_admin}",
+                "Content-type": "application/json",
+            },
+        )
+
+        assert r.status_code == 422
+        assert r.json() == {
+            "detail": [
+                {
+                    "loc": ["body"],
+                    "msg": "QueueFreezePayload.__init__() got an unexpected keyword argument 'false_key'",
```

Comment:
Yes, I just reported the payload that fastapi+pydantic are sending back when this error case is hit. We can manage to change it though !

Reply:
We really should. This really does not look like an error a user would be happy to see, it really makes me thing there's an uncaught error on the *server-side* rather than a problem on the *client-side*.

Reply:
I added an exception handler for those cases in order to not sendback the fastapi+pydantic error payload, light need some improvement in the verbose but it seems more appropriate like this